weatherman.py

```diff
@@ -0,0 +1,246 @@
+            __weather__[city] = {}
+        if year not in __weather__[city].keys():
+            __weather__[city][year] = {}
+        if month not in __weather__[city][year].keys():
+            __weather__[city][year][month] = {}
+        # reading file content
+        file = open(file_name, 'r')
+        lines = file.readlines()
+
+        for line in range(1, len(lines)):
+            info = lines[line].split(",")
+            details = {}
+            details.update({"PKT": info[0]})
+            details.update({"Max TemperatureC": int(info[1]) if info[1] != '' else None})
+            details.update({"Mean TemperatureC": int(info[2])if info[2] != '' else None})
+            details.update({"Min TemperatureC": int(info[3]) if info[3] != '' else None})
+            details.update({"Dew PointC": int(info[4]) if info[4] != '' else None})
+            details.update({"MeanDew PointC": int(info[5]) if info[5] != '' else None})
+            details.update({"Min DewpointC": int(info[6]) if info[6] != '' else None})
+            details.update({"Max Humidity": int(info[7]) if info[7] != '' else None})
+            details.update({"Mean Humidity": int(info[8]) if info[8] != '' else None})
+            details.update({"Min Humidity": int(info[9]) if info[9] != '' else None})
+            details.update({"Max Sea Level PressurehPa": info[10]})
+            details.update({"Mean Sea Level PressurehPa": info[11]})
+            details.update({"Min Sea Level PressurehPa": info[12]})
+            details.update({"Max VisibilityKm": info[13]})
+            details.update({"Mean VisibilityKm": info[14]})
+            details.update({"Min VisibilitykM": info[15]})
+            details.update({"Max Wind SpeedKm/h": info[16]})
+            details.update({"Max Gust SpeedKm/h": info[17]})
+            details.update({"Precipitationmm": info[18]})
+            details.update({"CloudCover": info[19]})
+            details.update({"Events": info[20]})
+            details.update({"WindDirDegrees": info[21]})
+            __weather__[city][year][month][line] = details
+
+    def read_folder(self, directory_path):
+        files = os.listdir(directory_path)
+        # file format should be city_weather_year_month.txt for proper parsing
+        for file in files:
+            self.read_file(directory_path+"/"+file)
+
+
+class Computations:
+
+    @staticmethod
+    def maximum_temperature(year):
+        maximum_temp_month = "Jan"
+        maximum_temp = __weather__["Murree"][year][maximum_temp_month][1]["Max TemperatureC"]
+        maximum_temp_day = 1
+        for month in __weather__["Murree"][year].keys():
+            for day in __weather__["Murree"][year][month].keys():
+                if not maximum_temp:
+                    maximum_temp = __weather__["Murree"][year][month][day]["Max TemperatureC"]
+                if (__weather__["Murree"][year][month][day]["Max TemperatureC"] and __weather__["Murree"][year][month][day]["Max TemperatureC"] >
```

Comment:
PEP8 violation. No line should exceed 79 characters but we can allow it till 118 but not more than that.